src/v/cloud_storage/remote.cc

```diff
@@ -1268,6 +1268,8 @@ ss::future<upload_result> remote::delete_objects_sequentially(
                              results.push_back(result);
                          });
                    })
+            .handle_exception_type(
+              [](const ss::abort_requested_exception& /* ignore */) {})
```

Comment:
nit: it might be useful to still log these, but at `debug` log level. In digging into CI failures, silently dropped exceptions can be the difference between an obvious root cause or not (note that most CI runs with `trace` level logging enabled)

Reply:
Ahh good point. I advised @jcipar to not log at all (based on ssx::background behavior) but I think you're right.